planner/core/plan_stats.go

```diff
@@ -170,3 +172,95 @@ func collectHistNeededItems(histNeededColumns []model.TableItemID, histNeededInd
 	histNeededItems = append(histNeededItems, histNeededColumns...)
 	return
 }
+
+func recordTableRuntimeStatsJSON(sctx sessionctx.Context, tbls map[int64]struct{}) {
```

Comment:
Better to collect/record these StatsJSONs lazily since this process is relatively heavy and it seems that not each query needs these StatsJSONs.

Reply:
updated.